src/realm/sync/transform.cpp

```diff
@@ -2591,7 +2591,8 @@ Changeset& Transformer::get_reciprocal_transform(TransformHistory& history, file
                                                  version_type version, const HistoryEntry& history_entry)
 {
     auto& changeset = m_reciprocal_transform_cache[version]; // Throws
-    if (changeset.empty()) {
+    // There can be empty changesets in the cache, so check the version too.
+    if (changeset.empty() && changeset.version == 0) {
```

Comment:
remind me, how do you end up with a changeset in the cache where the version is zero?

Reply:
the instruction above inserts an empty changeset (with version zero) if the key is not found. subtle, but one the culprits.

Reply:
Would you mind updating the comment to explain this? Thanks

Reply:
interesting. is there a reason why we can't use cache.insert right above rather than the subscript operator so we actually get the inserted flag rather than checking if the version is zero?